src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends { [key: string]: number | string }>(
+  items: T[],
+  key: string,
+): GroupsMap<T> {
+  return items.reduce((result: GroupsMap<T>, item: T) => {
+    if (!result[item[key]]) {
+      result[item[key]] = [];
+    }
```

Comment:
Using bracket notation with a string or number key can lead to potential type errors if `item[key]` is not a string. To ensure type safety, consider using a type guard to check that `item[key]` is indeed a string before using it as a property key.